vispy/visuals/image.py

```diff
@@ -208,12 +232,20 @@ class ImageVisual(Visual):
 
     VERTEX_SHADER = VERT_SHADER
     FRAGMENT_SHADER = FRAG_SHADER
+    COMPLEX_MODES = set(COMPLEX_TRANSFORMS)
 
     def __init__(self, data=None, method='auto', grid=(1, 1),
                  cmap='viridis', clim='auto', gamma=1.0,
-                 interpolation='nearest', texture_format=None, **kwargs):
+                 interpolation='nearest', texture_format=None, complex_mode="magnitude",
+                 **kwargs):
         """Initialize image properties, texture storage, and interpolation methods."""
         self._data = None
+        self._data_is_complex = False
+        if complex_mode not in self.COMPLEX_MODES:
+            raise ValueError(
+                "complex_mode must be one of %s" % ", ".join(self.COMPLEX_MODES)
+            )
+        self._complex_mode = complex_mode
```

Comment:
Can this be replaced with `self.complex_mode = complex_mode` (maybe a `self._complex_mode = None` first. That way the error message isn't repeated.